sympy/functions/elementary/complexes.py

```diff
@@ -140,12 +166,30 @@ class im(Function):
     >>> from sympy.abc import x, y
     >>> im(2*E)
     0
-    >>> re(2*I + 17)
-    17
+    >>> im(2*I + 17)
+    2
     >>> im(x*I)
     re(x)
     >>> im(re(x) + y)
     im(y)
+    >>> im(x + y*I)
+    re(y) + im(x)
+    >>> im(2 + 3*I).as_real_imag()
```

Comment:
What is the purpose of this example?

Reply:
To demonstrate use of as_real_imag()

Reply:
But this is the docstring for `im`. `im(x).as_real_imag()` will always return `(0, im(x))`, so there's no reason why anyone would need to call as_real_imag() after using `im`.